runtime/server_http_response.go

```diff
@@ -291,7 +292,9 @@ func (res *ServerHTTPResponse) flush(ctx context.Context) {
 
 	res.flushed = true
 	res.writeHeader(res.pendingStatusCode)
-	res.writeBytes(res.pendingBodyBytes)
+	if res.pendingStatusCode != http.StatusNoContent {
```

Comment:
we use 304 for use cases so you probably want to accommodate for that status code as well.